src/pages/index.tsx

```diff
@@ -4,8 +4,13 @@ import Head from "next/head";
 import { api } from "~/utils/api";
 
 export default function Home() {
-  const hello = api.example.hello.useQuery({ text: "from tRPC" });
   const user = useUser();
+
+  //to fetch all data from DB
+  const {data, error, isLoading} = api.posts.getAll.useQuery();
+
+  const renderData = data?.map((post) => (<div key={post.id}>{post.content}</div>))
```

Comment:
```suggestion
  if(!data) throw new Error(' fetch is failed. data isn't found ')
  const renderData = data.map((post) => (<div key={post.id}>{post.content}</div>))
```

型エラーをなくす方法の一つとして、使用する前に`if`でCheckすることができます！
これにより`Undefined`をクリアすることができると思います！

参考になりそうな記事も下記に記載しておきます！
[early return ](https://stackoverflow.com/questions/76847076/are-early-return-typeguards-possible-in-typescript)